### ISSUE TITLE
Loss function

### ISSUE THREAD

**Issue by awmacpherson** [shtukaresearch/livepeer-emissions-risk#8]
**Is your feature request related to a problem? Please describe.**
We currently have no quantitative criterion to rate models against one another.

**Describe the solution you'd like**
We'll define a handcrafted loss function. Here's my attempt:

First, let $T \subset\mathbb{N}$ be the set of forecast epochs. If we're forecasting average participation rate for each calendar month from December until June next year inclusive, $|T| = 8$. If $\phi$ is a predicate on $T$, define the *visit set* 

$$
T_\phi := \\{t\in T \mid \phi(t)\\}.
$$
 
We're interested in predicates defined in terms of projected values $\tilde{P}_t$ for $t\in T$. Each $\tilde{P}_t$ is a real valued random variable. 

In particular, we introduce (random) predicates $\phi_P$ indexed by $k\in \mathrm{N}$ defined by

$$
\phi_{P}(t) \quad \Leftrightarrow |\tilde{P}_t < P |. 
$$ 

The associated (random) subset $T_P := T_{\phi_P}$ is called the *breach set* — the set of epochs for which the random process $\tilde{P}_\bullet$ breaches the lower threshold $P$. The cardinalities of these sets $\tilde{K}_P :=|T_P|$ are the *breach counts.* The breach counts define a family of distribution functions $F_P(k) = \mathrm{Prob}(\tilde{K}_P\leq k)$. The family is decreasing in $P$: if we move the threshold up, breach counts go up, so the CDF shifts to the right.

Introduce a family of "quantile numbers" $P_{q,k}\in [0,1]$ parametrised by $q\in[0,1]$ and $k\in\mathbb{N}$ defined by inverting $F_P(k)$. In other words,

$$
P_{q,k} = \sup\\{ P > 0 \mid F_P(k) \geq q \\}.
$$

In words, $P_{q,k}$ is the highest threshold $P>0$ whose breach count is $\leq k$ with probability at least $q$.

Now for the loss function. We want to penalise forecasts that underestimate breach counts. For each model, we run cross-validation over different train and validation windows of the data. For each train/validate set, we:
* Calculate $P_{q,0}$ and $P_{q,1}$ for given "high" values of $q$ (probably $q=0.95$ and $q=0.99$).
* Dock a point for each run where the validation data breaches $P_{q,0}$ one or more times.
* Dock a point for each run where the validation data breaches $P_{q,1}$ two or more times.

The loss $\mathcal{L}_q$ of that model is obtained by adding up the scores over all runs. It is a non-positive integer.

**Describe alternatives you've considered**
* MSE: We don't care about point forecasts. We don't want to penalise "errors" that don't affect the estimated probability of risk events.
* AIC or Akaike weights: we can also use these.
* We could also consider penalising models that *overestimate* breach counts, or penalise models on their forecasting of emissions, but that would be more complicated.
